File: 프로그래머스/1/178871. 달리기 경주/달리기 경주.py
def solution(players, callings):
    pla_dic = {player: i for i , player in enumerate(players)}
    
    for p in callings: 
        i = pla_dic[p]
        pla_dic[p] -= 1
        pla_dic[players[i-1]] +=1
        players[i-1], players[i] =  players[i], players[i-1] 
    return players

File: 프로그래머스/1/178871. 달리기 경주/달리기 경주.js
function solution(players, callings) {
    var answer = [];
    let dic = {};
    for (let i in players) {
        dic[players[i]] = i;
    }
        
    
    for (let player of callings){
        n = dic[player]
        dic[player] = n-1
        dic[players[n-1]] = n
        c = players[n]
        players[n] = players[n-1]
        players[n-1] = c
    }
        
    
    return players;
}